### ISSUE TITLE
iOS 8 grouped fetched results controller freezes

### ISSUE THREAD

**Issue by sigmundfridge** [magicalpanda/MagicalRecord#815]
I raised this as a more generic iOS 8 question previously (#813) but I think it makes sense to reopen it as a specific issue.

I have found that MR 2.3 (and below) freezes up in iOS 8. Steps to replicate:
- Sequentially create two FRCs using MR_fetchAllGroupedBy:@"aValue" ...... and it freezes on creating the second
- Setting the first FRC to nil before creating the second does not make a difference.
- The custom view controller container I am using freezes if navigating from VC to VC where a single grouped FRC for the same entity is used (similar to a replace segue, rather than a pop/push). Using a basic navigation controller solves this, so the timings of viewDidUnload and dealloc could be important.

Things I have noticed:
- Grouping appears to be the issue. Creating 2 FRCs gives the following:

| FRC 1 grouping | FRC2  grouping | Result |
| --- | --- | --- |
| nil | nil | No freeze |
| nil | string | Freeze |
| string | string | Freeze |
| string | nil | No freeze |
- changing delegates makes no difference
- the freeze appears to corrupt the database
- There is no freeze when manually creating FRCs (initWithFetchRequest.....)

So in summary MR 2.3 freezes on iOS8 when creating a grouped FRC whenever a FRC has previously been created for the same NSManagedObject entity (even if it has since been set as nil). 

Example basic project showing the freeze can be found here
https://github.com/sigmundfridge/MR_iOS8_freeze
